test/__snapshots__/v3-legacy-enums/test/generated/v3_legacy_enums/core/OpenAPI.ts.snap

```diff
@@ -1,8 +1,34 @@
 import type { ApiRequestOptions } from './ApiRequestOptions';
-import type { TConfig, TResult } from './types';
+import type { TResult } from './types';
 
-type Resolver<T> = (options: ApiRequestOptions) => Promise<T>;
 type Headers = Record<string, string>;
+type Middleware<T> = (value: T) => T | Promise<T>;
+type Resolver<T> = (options: ApiRequestOptions) => Promise<T>;
+
+interface Interceptor<T> {
+    _fns: ReadonlyArray<Middleware<T>>;
+    eject: (fn: Middleware<T>) => void;
+    use: (fn: Middleware<T>) => void;
+}
+
+export class Interceptors<T> {
+    _fns: Middleware<T>[];
```

Comment:
make this `public fns` since it is public (no `_`)

Reply:
I don't want to encourage using this field directly. It should be private ideally, the only methods users should interact with are use() and eject()

Reply:
ok, lets leave it as is for now